zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/validator/ProcessInstanceRequestValidator.java

```diff
@@ -33,11 +34,26 @@ public static Optional<ProblemDetail> validateCreateProcessInstanceRequest(
                 ERROR_MESSAGE_ONLY_ONE_FIELD.formatted(
                     List.of("bpmnProcessId", "processDefinitionKey")));
           }
-          if (request.getOperationReference() != null && request.getOperationReference() < 1) {
-            violations.add(
-                ERROR_MESSAGE_INVALID_ATTRIBUTE_VALUE.formatted(
-                    "operationReference", request.getOperationReference(), "> 0"));
+          validateOperationReference(request.getOperationReference(), violations);
+        });
+  }
+
+  public static Optional<ProblemDetail> validateCancelProcessInstanceRequest(
+      final CancelProcessInstanceRequest request) {
+    return validate(
+        violations -> {
+          if (request != null) {
+            validateOperationReference(request.getOperationReference(), violations);
           }
         });
   }
+
+  private static void validateOperationReference(
```

Comment:
🔧 There it is! Finally! 😄 I guess we can also move this to the general validator utility class and reuse this a bit in the other validators? Can also be a follow-up but it's a small change, I guess. I'll leave that to you to decide.

Reply:
I'll refactor this in a separate PR

Reply:
👍